include/dlaf/eigensolver/gen_eigensolver/impl.h

```diff
@@ -40,15 +46,39 @@ template <Backend B, Device D, class T>
 void GenEigensolver<B, D, T>::call(comm::CommunicatorGrid& grid, blas::Uplo uplo, Matrix<T, D>& mat_a,
                                    Matrix<T, D>& mat_b, Matrix<BaseType<T>, D>& eigenvalues,
                                    Matrix<T, D>& eigenvectors, const Factorization factorization) {
+#ifdef DLAF_WITH_HDF5
+  static std::atomic<size_t> num_gen_eigensolver_calls = 0;
+  std::stringstream fname;
+  fname << "generalized-eigensolver-" << matrix::internal::TypeToString_v<T> << "-"
+        << std::to_string(num_gen_eigensolver_calls) << ".h5";
+  std::optional<matrix::internal::FileHDF5> file;
+
+  if (getTuneParameters().debug_dump_generalized_eigensolver_data) {
+    file = matrix::internal::FileHDF5(grid.fullCommunicator(), fname.str());
+    file->write(mat_a, "/input-a");
+    file->write(mat_b, "/input-b");
```

Comment:
And maybe we can name it `/input_b` or `/input-b-factorized` depending on `factorization`. What do you think?